client/client.go

```diff
@@ -386,6 +386,60 @@ func (c *Client) BalanceProofWatcher() {
 	}
 }
 
+// FrozenWatcher listenes for Frozen events and calls WithdrawFrozen
+// when a balance proof is available.
+func (c *Client) FrozenWatcher() {
+	ctx, cancel := context.WithCancel(c.Ctx())
+	defer cancel()
+	sub, err := c.ethClient.SubscribeFrozen(ctx, c.contract, nil)
+	if err != nil {
+		c.logError("Frozen event subscription: %v", err)
+		return
+	}
+	defer sub.Unsubscribe()
+
+	select {
+	case err := <-sub.Err():
+		c.logError("Frozen event subscription: %v", err)
+		return
+	case event := <-sub.Events():
+		c.handleFrozen(event)
+	}
+}
+
+func (c *Client) handleFrozen(event *bindings.ErdstallFrozen) {
+	epoch := event.Epoch
+	c.events <- &Event{Type: SET_OP_TRUST, OpTrust: UNTRUSTED}
```

Comment:
you may also want to create a method for setting the op trust, like you did for the other `Event`s

Reply:
Added `setOpTrust`.